Exiled.Events/Patches/Events/Player/FirearmRequestReceived.cs

```diff
@@ -263,6 +265,34 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
                     new(OpCodes.Stloc_S, 6),
                 });
 
+            offset = -3;
+            index = newInstructions.FindLastIndex(instruction => instruction.opcode == OpCodes.Ldftn) + offset;
+
+            newInstructions.InsertRange(
+                index,
+                new[]
+                {
+                    // player
+                    new CodeInstruction(OpCodes.Ldloc_S, player.LocalIndex),
+
+                    // firearm
+                    new(OpCodes.Ldloc_S, firearm.LocalIndex),
+
+                    // InspectingWeaponEventArgs ev = new(Player, firearm)
+                    new(OpCodes.Newobj, GetDeclaredConstructors(typeof(InspectingWeaponEventArgs))[0]),
+                    new(OpCodes.Dup),
+                    new(OpCodes.Dup),
+                    new(OpCodes.Stloc_S, ev.LocalIndex),
+
+                    // Player.OnInspectingWeapon(ev)
+                    new(OpCodes.Call, Method(typeof(Player), nameof(Player.InspectingWeapon))),
+
+                    // if (!ev.IsAllowed)
+                    //    return;
+                    new(OpCodes.Callvirt, PropertyGetter(typeof(InspectingWeaponEventArgs), nameof(InspectingWeaponEventArgs.IsAllowed))),
+                    new(OpCodes.Brfalse_S, returnLabel),
+                });
+
             newInstructions[newInstructions.Count - 1].WithLabels(returnLabel);
```

Comment:
`[...].labels.Add(returnLabel)`

Reply:
yeah thats how it did it in the base code, but i can change it rq